decorator.rb

```diff
@@ -0,0 +1,30 @@
+require_relative './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
+  end
+
+  attr_accessor :nameable
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    if @nameable.correct_name.length >= 10
+      @nameable.correct_name.slice(0, 10)
+    else
+      @nameable.correct_name
+    end
+  end
+end
```

Comment:
[OPTIONAL] Please move the `CapitalizeDecorator` and `TrimmerDecorator` classes to their own files. Although they both inherit from the `Decorator` class, it's not a best practice to have 3 classes with different functionalities in the same file.

Reply:
@KDlamini Thank you. I'll surely implement it now.